cmd/gnodev/mod.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/gnolang/gno/pkgs/command"
+	"github.com/gnolang/gno/pkgs/errors"
+	"github.com/gnolang/gno/pkgs/gnolang/gnomod"
+)
+
+type modFlags struct {
+	Verbose bool `flag:"verbose" help:"verbose"`
+}
+
+var defaultModFlags = modFlags{
+	Verbose: false,
+}
+
+func modApp(cmd *command.Command, args []string, iopts interface{}) error {
+	opts := iopts.(modFlags)
+
+	if len(args) != 1 {
+		cmd.ErrPrintfln("Usage: mod [flags] <command>")
+		return errors.New("invalid command")
+	}
+
+	switch args[0] {
+	case "download":
+		if err := runModDownload(&opts); err != nil {
+			return fmt.Errorf("mod download: %w", err)
+		}
+	default:
+		return fmt.Errorf("invalid command: %s", args[0])
+	}
+
+	return nil
+}
+
+func runModDownload(opts *modFlags) error {
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	modPath := filepath.Join(path, "gno.mod")
+	if !isFileExist(modPath) {
+		return errors.New("gno.mod not found")
+	}
+
+	// read gno.mod
+	data, err := os.ReadFile(modPath)
+	if err != nil {
+		return fmt.Errorf("readfile %q: %w", modPath, err)
+	}
+
+	// parse gno.mod
+	gnoMod, err := gnomod.Parse(modPath, data)
+	if err != nil {
+		return fmt.Errorf("parse: %w", err)
+	}
+
+	// fetch dependencies
+	if err := gnoMod.FetchDeps(); err != nil {
+		return fmt.Errorf("fetch: %w", err)
+	}
+
+	if err := gnomod.Sanitize(gnoMod); err != nil {
+		return fmt.Errorf("sanitize: %w", err)
+	}
+
+	// write go.mod file
+	err = gnoMod.WriteToPath(path)
+	if err != nil {
+		return fmt.Errorf("write go.mod file: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("write gno.mod file: %w", err)
```

Reply:
I think it's correct. It creates a temp go.mod file based on gno.mod. (Just like *gno.gen.go)

Reply:
oh ok, so maybe gno.gen.mod, so we can also have a go.mod in the same folder for other reasons?

Reply:
I don't think `go` recognizes any filename other than `go.mod`. `gno.gen.mod` is not possible.

In the future, If we decided to have `go.mod` in the same folder that destroys the whole purpose of `gno.mod`. Anyways, we can always include stuff in `gno.mod` that we want to retain in generated `go.mod`.

Reply:
in the future, probably the generated files won't be in the same folder but in another one or in memory, so the current folder only contains "current" files.

so let's discuss this later.